File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);


	vector <string> stack;
	int N;
	cin >> N;
	cin.ignore();

	while (N--) {
		string command;
		getline(cin, command);
		if (command.substr(0, 4) == "push") {
			string to_push = command.substr(5, command.size() - 1);
			stack.push_back(to_push);
		}
		else if (command == "pop") {
			if (stack.empty()) cout << -1 << "\n";
			else {
				cout << stack.back() << "\n";
				stack.pop_back();
			}
		}
		else if (command == "size") cout << stack.size() << "\n";
		else if (command == "empty") {
			if (stack.empty()) cout << 1 << "\n";
			else cout << 0 << "\n";
		}
		else if (command == "top") {
			if (stack.empty()) cout << -1 << "\n";
			else cout << stack.back() << "\n";
		}
	}

	return 0;
}

File: 백준/Silver/10828. 스택/스택.py
import sys

input = sys.stdin.readline

N_str = input().strip()
if N_str:
    N=int(N_str)
    stack=[]

    for _ in range(N):
        command = input().split()
        if command[0] == "push":
            num = command[1]
            stack.append(num)
        elif command[0] == "pop":
            if stack:
                print(stack.pop())
            else:
                print(-1)
        elif command[0] == "size":
            print(len(stack))
        elif command[0] == "empty":
            if len(stack) == 0:
                print(1)
            else:
                print(0)
        elif command[0] == "top":
            if len(stack)!=0:
                print(stack[-1])
            else:
                print(-1)